File: csharp-pr/47 2/csharp/code-quality-error-handling/NetworkClient.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

namespace NetworkServices
{
    public class NetworkClient
    {
        private HttpClient httpClient = new HttpClient();

        public async Task<string> FetchDataAsync(string url)
        {
            // {fact rule=code-quality-error-handling@v1.0 defects=1}
            try
            {
                var response = await httpClient.GetAsync(url);
                return await response.Content.ReadAsStringAsync();
            }
            catch
            {
                throw new Exception("Something went wrong");
            }
            // {/fact}
        }
    }
}

File: cpp-pr/46/cpp/code-quality-error-handling/network_client.cpp
#include <iostream>
#include <string>
#include <vector>

class NetworkClient {
public:
    std::string sendRequest(const std::string& url, const std::string& data) {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        std::string response = "HTTP/1.1 200 OK\nContent: " + data;
        return response;
        // {/fact}
    }
    
    std::vector<std::string> downloadFiles(const std::vector<std::string>& urls) {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        std::vector<std::string> results;
        for (const auto& url : urls) {
            try {
                if (url.empty()) {
                    throw std::invalid_argument("URL cannot be empty");
                }
                results.push_back("Downloaded: " + url);
            } catch (const std::exception& e) {
                std::cerr << "Error downloading " << url << ": " << e.what() << std::endl;
                results.push_back("Failed: " + url);
            }
        }
        return results;
        // {/fact}
    }
};

int main() {
    NetworkClient client;
    std::string response = client.sendRequest("http://api.example.com", "test data");
    std::cout << response << std::endl;
    return 0;
}